src/components/IsOrderOwner.js

```diff
@@ -0,0 +1,46 @@
+import { useState, useEffect, useContext } from 'react';
+import { useParams, Navigate } from 'react-router-dom';
+import { useSelector } from 'react-redux';
+import { AuthContext } from '../context/auth.context';
+import { Box, CircularProgress } from '@mui/material';
+import SuccessMessage from './SuccessMessage';
+
+export const IsOrderOwner = ({ children }) => {
+  const { isLoggedIn, isLoading, user } = useContext(AuthContext);
+  const { shopOrders } = useSelector((store) => store.orders);
+  const [order, setOrder] = useState(null);
+  const { orderId } = useParams();
+
+  let messageToDisplay = () => {
+    if (user._id !== order?.userId._id) {
+      return 'Utilizador inválido.';
+    } else {
+      return 'Review para este pedido já foi submetido. Obrigado.';
+    }
+  };
+
+  useEffect(() => {
+    if (shopOrders.length > 0 && orderId) {
+      let orderToReview = shopOrders.find((item) => item._id === orderId);
+      setOrder(orderToReview);
+    }
+  }, [shopOrders, orderId]);
+
+  // If the authentication is still loading
+  if (isLoading) return <CircularProgress sx={{ mt: 20 }} size={80} />;
+
+  if (isLoggedIn && user._id === order?.userId._id && !order?.reviewId) {
+    // If the user is logged in, allow to see the page
+    return children;
+  }
+  if (!isLoggedIn) {
```

Comment:
You can create a global hook that checks if you're logged in and then redirects if you're not

Reply:
Just not so sure because I already have `isLoggedIn` to know that. And in this file is more about protecting the access to the page based on a very specific condition. But I may be not understanding your point 100%, can you explain more?